.github/workflows/ci.yml

```
@@ -388,6 +388,71 @@ jobs:
       run: cmake --build build --target test_cmake_build
 
 
+  # Testing on ICX/ICPX, Intel's next-gen C/C++ compiler using the oneAPI apt repo
+  icc:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+
+    name: "🐍 3 • ICPX latest • x64"
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Add apt repo
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y wget build-essential pkg-config cmake ca-certificates gnupg
+        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+        echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+    - name: Add ICC & Python 3
+      run: sudo apt-get update; sudo apt-get install -y intel-oneapi-compiler-dpcpp-cpp cmake python3-dev python3-numpy python3-pytest python3-pip
+
+    - name: Update pip
+      shell: bash
+      run: |
+        set +e; source /opt/intel/oneapi/setvars.sh; set -e
+        python3 -m pip install --upgrade pip
+    - name: Install dependencies
+        set +e; source /opt/intel/oneapi/setvars.sh; set -e
+        python3 -m pip install -r tests/requirements.txt --prefer-binary
+
+    - name: Configure
+      shell: bash
+      run: |
+        set +e; source /opt/intel/oneapi/setvars.sh; set -e
+        cmake -S . -B build     \
+        -DPYBIND11_WERROR=ON    \
+        -DDOWNLOAD_CATCH=ON     \
+        -DDOWNLOAD_EIGEN=OFF    \
+        -DCMAKE_CXX_STANDARD=11             \
+        -DCMAKE_CXX_COMPILER=$(which icpx)  \
+        -DCMAKE_VERBOSE_MAKEFILE=ON         \
+        -DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)")
```

Temporarily needs this patch since an Intel contributor broke CMake mainline recently, which is currently shipped in oneAPI (breakage was reverted in mainline CMake already):
```suggestion
        -DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)") \
        -DCMAKE_CXX_COMPILER_ID="Clang"                \
        -DCMAKE_CXX_COMPILER_VERSION=12.0              \
        -DCMAKE_CXX_STANDARD_COMPUTED_DEFAULT="17"     \
```